### PR TITLE
better compose behaviour for empty/single function compositions

### PR DESCRIPTION
**Pull request by nitzo**

### CODE REVIEW

**Review by tuti-lv — APPROVED** [review]
notice the build had failed

**Comment by nitzo** [comment]
@tuti-lv it's the docs build job. We have an issue with it

File: gamla/functional_generic.py [hunk 1]
```diff
@@ -130,6 +131,12 @@ def compose(*funcs):
     #         logging.error(f"composing {prev} after {f}, which don't fit")
     #     prev = f
 
+    if operator.empty(funcs):
+        return gamla.identity
+
+    if len(funcs) == 1:
+        return operator.head(funcs)
```
Comment (uriva) [inline gamla/functional_generic.py:138]: @nitzo remember this is a very sensitive place performance wise, probably funcs[0] is faster than an additional function call
similarly `not funcs` will be faster than `operator.empty(funcs)`

File: gamla/functional_generic.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
     Union,
 )
 
+import gamla
```
Comment (uriva) [inline gamla/functional_generic.py:21]: @nitzo we import from the internal dirs in the lib itself

**Pull request merged**